### PR TITLE
Patch Mcore load_state_dict at runtime for TE 1.14

### PR DESCRIPTION
**Pull request by chtruong814**
# What does this PR do ?

Attempt to patch the load_state_dict method for Mcore optimizer for TE 1.14 to address a bug where state values may be None.

**Collection**: [Note which collection this PR will affect]

# Changelog

- Patch Mcore load_state_dict at runtime for TE 1.14

# Usage

n/a

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

**Comment by chtruong814** [comment]
Just wanting to see if this passes one test first. If that somehow actually works, will ask for feedback from other Devs.  We may want to do this in other places or push further upstream into Mcore itself.

**Comment by chtruong814** [comment]
Unfortunately this didn't do the job but we have another workaround happening
https://github.com/NVIDIA/NeMo/actions/runs/13334143243/job/37249723187
